jose/Jws.ml

```diff
@@ -20,31 +20,32 @@ let to_string t =
   |> RResult.map (fun (header_str, payload_str) ->
          header_str ^ "." ^ payload_str ^ "." ^ t.signature)
 
-let verify_RS256 ~jwk str =
-  match jwk with
-  | Jwk.Pub.RSA jwk ->
-      Jwk.Pub.rsa_to_pub jwk
-      |> RResult.map (fun key -> Mirage_crypto_pk.Rsa.PKCS1.sig_decode ~key str)
-      |> RResult.flat_map (function
-           | None -> Error (`Msg "Could not decode signature")
-           | Some message -> Ok message)
-  | _ -> Error (`Msg "JWK doesn't match")
+let verify_RS256 (type a) ~(jwk : a Jwk.t) str =
+  ( match jwk with
+  | Jwk.Rsa_priv jwk ->
+      Jwk.pub_of_priv_rsa jwk |> fun jwk ->
+      Mirage_crypto_pk.Rsa.PKCS1.sig_decode ~key:jwk.key str
+  | Jwk.Rsa_pub jwk -> Mirage_crypto_pk.Rsa.PKCS1.sig_decode ~key:jwk.key str
+  | _ -> None )
```

Comment:
I'd list `Oct` specifically so that we can trigger warning 8 if / when you add more variant constructors.